stock_shipment_management/model/shipment_plan.py

```diff
@@ -0,0 +1,354 @@
+        track_visibility='onchange',
+        help="Up to date Estimated Time of Arrival"
+    )
+    from_address_id = fields.Many2one(
+        'res.partner', 'From Address',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        required=True,
+    )
+    to_address_id = fields.Many2one(
+        'res.partner',
+        'To Address',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        required=True,
+    )
+    consignee_id = fields.Many2one(
+        'res.partner',
+        'Consignee',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    carrier_tracking_ref = fields.Char(
+        'Tracking Ref.',
+        readonly=True,  # updated by wizard
+        track_visibility='onchange',
+    )
+    transport_mode_id = fields.Many2one(
+        'transport.mode',
+        string='Transport by',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    note = fields.Text(
+        'Description / Remarks',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        [('draft', 'Draft'),
+         ('confirmed', 'Confirmed'),
+         ('in_transit', 'In Transit'),
+         ('done', 'Done'),
+         ('cancel', 'Cancel')
+         ],
+        required=True,
+        default='draft',
+        copy=False,
+        track_visibility='onchange',
+    )
+    departure_move_ids = fields.One2many(
+        'stock.move', 'departure_shipment_id', 'Departure Moves',
+        readonly=True, copy=False,
+    )
+    arrival_move_ids = fields.One2many(
+        'stock.move', 'arrival_shipment_id', 'Arrival Moves',
+        readonly=True, copy=False,
+    )
+    departure_picking_ids = fields.One2many(
+        compute='_get_departure_picking_ids',
+        comodel_name='stock.picking',
+        string='Departure Pickings',
+        readonly=True, copy=False,
+    )
+    arrival_picking_ids = fields.One2many(
+        compute='_get_arrival_picking_ids',
+        comodel_name='stock.picking',
+        string='Arrival Pickings',
+        readonly=True, copy=False,
+    )
+    volume = fields.Float(
+        compute='_compute_volume',
+        readonly=True,
+        store=True,
+    )
+    weight = fields.Float(
+        compute='_compute_weights',
+        readonly=True,
+        store=True,
+    )
+    weight_net = fields.Float(
+        compute='_compute_weights',
+        readonly=True,
+    )
+    departure_picking_count = fields.Integer(
+        compute='_departure_picking_count',
+    )
+    arrival_picking_count = fields.Integer(
+        compute='_arrival_picking_count',
+    )
+
+    @api.one
+    @api.constrains('initial_etd', 'initial_eta')
+    def _check_initial_estimated_times(self):
+        if (self.initial_etd and self.initial_eta and
+                self.initial_etd > self.initial_eta):
+            raise Warning(_('Initial ETD cannot be set after initial ETD.'))
```

Comment:
According to the documentation [0], if the validation failed, a `ValidationError` should be raised.

[0] https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.api.constrains